src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-storage-class-form/rgw-storage-class-form.component.ts

```diff
@@ -150,78 +165,125 @@ export class RgwStorageClassFormComponent extends CdForm implements OnInit {
     this.createForm();
     this.storageClassTypeText();
     this.updateTierTypeHelpText();
-    this.loadingReady();
     this.loadZoneGroup();
     if (this.editing) {
       this.route.params.subscribe((params: StorageClass) => {
         this.storageClassInfo = params;
       });
       this.rgwStorageService
         .getPlacement_target(this.storageClassInfo.placement_target)
-        .subscribe((placementTargetInfo: PlacementTarget) => {
-          this.tierTargetInfo = this.getTierTargetByStorageClass(
-            placementTargetInfo,
-            this.storageClassInfo.storage_class
-          );
-          let response = this.tierTargetInfo?.val?.s3;
-          const aclMappings = this.tierTargetInfo?.val?.s3?.acl_mappings || [];
-          this.storageClassForm.get('zonegroup').disable();
-          this.storageClassForm.get('placement_target').disable();
-          this.storageClassForm.get('storage_class').disable();
-          if (
-            this.tierTargetInfo?.val?.tier_type === TIER_TYPE.CLOUD_TIER ||
-            this.tierTargetInfo?.val?.tier_type === TIER_TYPE.GLACIER
-          ) {
+        .pipe(
+          switchMap((placementTargetInfo: PlacementTarget) => {
+            // Set the tierTargetInfo based on the placementTargetInfo and storageClassInfo
+            this.tierTargetInfo = this.getTierTargetByStorageClass(
+              placementTargetInfo,
+              this.storageClassInfo.storage_class
+            );
+            const tierType = this.tierTargetInfo?.val?.tier_type ?? TIER_TYPE.LOCAL;
+
+            // If tierType is LOCAL, make the second API calls
+            if (tierType === TIER_TYPE.LOCAL) {
+              return forkJoin([
+                this.poolService.getList(),
+                this.rgwZoneService.getAllZonesInfo()
+              ]).pipe(map(([pools, zones]) => ({ placementTargetInfo, pools, zones })));
+            }
+
+            // If tierType is not LOCAL, just return placementTargetInfo with null pools and zones
+            return of({ placementTargetInfo, pools: null, zones: null });
+          }),
+          map(({ placementTargetInfo, pools, zones }) => {
+            return { placementTargetInfo, pools, zones };
+          }),
+          catchError(() => {
+            return of({
+              placementTargetInfo: null,
+              pools: null,
+              zones: null
+            });
+          })
+        )
+        .subscribe(
+          (data: {
+            placementTargetInfo: PlacementTarget;
+            pools: Pool[] | null;
+            zones: AllZonesResponse | null;
+          }) => {
+            let response = this.tierTargetInfo?.val?.s3;
```

Comment:
NIT; In some places below `response` is being used and others `this.tierTargetInfo` makes it hard to read IMO